lib/matplotlib/quiver.py

```diff
@@ -513,58 +525,233 @@ def _init(self):
             trans = self._set_transform()
             self.span = trans.inverted().transform_bbox(self.axes.bbox).width
             if self.width is None:
-                sn = np.clip(math.sqrt(self.N), 8, 25)
+                sn = np.clip(math.sqrt(len(self.get_offsets())), 8, 25)
                 self.width = 0.06 * self.span / sn
 
             # _make_verts sets self.scale if not already specified
             if (self._dpi_at_last_init != self.axes.figure.dpi
                     and self.scale is None):
-                self._make_verts(self.XY, self.U, self.V, self.angles)
+                self._make_verts(self.get_offsets(), self._U, self._V, self.angles)
 
             self._dpi_at_last_init = self.axes.figure.dpi
 
+    @property
+    def N(self):
+        _api.warn_deprecated("3.9", alternative="get_X().size")
```

Comment:
All deprecations are missing the `name` argument. Also I thought @timhoffm suggested to not deprecate these? I don't know if that's been resolved.

Reply:
Yes, overall the state of the PR is not in the simplified form suggested in https://github.com/matplotlib/matplotlib/pull/26410#issuecomment-2017554032. As written there, it would best be done in a separate PR and exclusively add the `set_U()` etc. methods directly forwarding to the attribute `self.U`. None of the internal refactorings or `set_XYUVC()` is needed for that. Alternatively, we can skip that and do everything in this PR targeting 3.10.

Either way I‘m remilestoning this PR to 3.10. If you  still want the simple setter in 3.9, please open a separate PR for that.